src/components/data-exploration/heatmap/HeatmapPlot.jsx

```diff
@@ -206,33 +195,14 @@ const HeatmapPlot = (props) => {
     );
   }
 
-  if (selectedGenes?.length === 0) {
```

Comment:
We can not remove all genes from the heatmap, so this code path is never reached.

When removing genes, we filter the currently displayed genes against the list that we want to remove, creating a new list of genes that we want to display ([ref](https://github.com/hms-dbmi-cellenics/ui/blob/7180879ee2c35cb6ecf8282237ff5954edde9353/src/components/data-exploration/generic-gene-table/ComponentActions.jsx#L30-L34)). Removing all genes will result in an empty array, which is passed to `loadGeneExpression`. However, there is a check in `loadGeneExpression` which displays genes that are already loaded, in case an empty ilst of genes is passed to `loadGeneExpression` ([ref](https://github.com/hms-dbmi-cellenics/ui/blob/417c9d5fa11fc79f772824f5bae7ef880493d74b/src/redux/actions/genes/loadGeneExpression.js#L49-L59)).

Reply:
We can either :
- Modify `loadGeneExpression` to allow for no genes to be show in the heatmap (need a deeper exploration, maybe in another ticket)
- Or leave it just like this, if there's no problem.

Reply:
We should try to find out the reasoning behind that check, but instinctively I'd say we need to have the ability to remove all genes because it's what a user would expect